tests/modules/usercall.c

```diff
@@ -3,78 +3,61 @@
 RedisModuleUser *user = NULL;
 
 int call_without_user(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-    if (argc < 3) {
+    if (argc < 2) {
         return RedisModule_WrongArity(ctx);
     }
 
-    size_t cmd_len;
-    const char * cmd = RedisModule_StringPtrLen(argv[1], &cmd_len);
+    const char *cmd = RedisModule_StringPtrLen(argv[1], NULL);
 
-    RedisModuleCallReply *reply = RedisModule_Call(ctx, cmd, "Ev", argv + 2, argc - 2);
-    if (reply == NULL) {
-        RedisModule_ReplyWithError(ctx, "Failed to Execute");
-        return REDISMODULE_OK;
+    RedisModuleCallReply *rep = RedisModule_Call(ctx, cmd, "Ev", argv + 2, argc - 2);
+    if (!rep) {
+        RedisModule_ReplyWithError(ctx, "NULL reply returned");
+    } else {
+        RedisModule_ReplyWithCallReply(ctx, rep);
+        RedisModule_FreeCallReply(rep);
     }
-
-    RedisModule_ReplyWithCallReply(ctx, reply);
-    RedisModule_FreeCallReply(reply);
     return REDISMODULE_OK;
 }
 
 int call_with_user(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
```

Comment:
so now `call_with_user` is identical to `call_without_user`?
delete one?

Reply:
I dont think they are quite the same (but perhaps close enough).

call_without_user is normal RM_Call.

call_with_user is calling RM_SetContextModuleUser before RM_Call, but without any flags, so in practice similar, but more a test to ensure semantics dont change in future.

call_with_user_flags is calling RM_SetContextModuleUser and then whatever flag we use (presumably 'C') to make use of that call.

if you dont think its neccessary, can remove, but that was my logic.

Reply:
look again.
in the last version you posted call_with_user doesn't call RM_SetContextModuleUser, so it's identical to call_without_user.
also, if you want to test a case of using RM_SetContextModuleUser without the C flag, you can still do that with call_with_user_flags.